meilisearch/_local_server.py

```diff
@@ -0,0 +1,233 @@
+    ):
+        self.port = port or self._find_available_port()
+        self.data_path = data_path or tempfile.mkdtemp(prefix="meilisearch_")
+        self.master_key = master_key
+        self.process: Optional[subprocess.Popen] = None
+        self.binary_path: Optional[str] = None
+        self.url = f"http://127.0.0.1:{self.port}"
+        self._temp_data_dir = data_path is None
+
+    def _find_available_port(self) -> int:
+        """Find an available port to use."""
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.bind(("", 0))
+            return sock.getsockname()[1]
+
+    def _find_meilisearch_binary(self) -> Optional[str]:
+        """Find Meilisearch binary in PATH or common locations."""
+        # First, check if 'meilisearch' is in PATH
+        binary_name = "meilisearch" if platform.system() != "Windows" else "meilisearch.exe"
+        binary_path = shutil.which(binary_name)
+        if binary_path:
+            return binary_path
+
+        # Check common installation locations
+        common_paths = []
+        if platform.system() == "Darwin":  # macOS
+            common_paths.extend(
+                [
+                    "/usr/local/bin/meilisearch",
+                    "/opt/homebrew/bin/meilisearch",
+                    str(Path.home() / ".local" / "bin" / "meilisearch"),
+                ]
+            )
+        elif platform.system() == "Linux":
+            common_paths.extend(
+                [
+                    "/usr/local/bin/meilisearch",
+                    "/usr/bin/meilisearch",
+                    str(Path.home() / ".local" / "bin" / "meilisearch"),
+                ]
+            )
+
+        for path in common_paths:
+            if Path(path).exists() and Path(path).is_file():
+                return path
+
+        return None
+
+    def _download_meilisearch(self) -> str:
+        """Download Meilisearch binary for the current platform."""
+        system = platform.system().lower()
+        machine = platform.machine().lower()
+
+        # Map platform to Meilisearch release names
+        if system == "darwin":
+            if machine in ["arm64", "aarch64"]:
+                platform_str = "apple-silicon"
+            else:
+                platform_str = "amd64"
+            binary_name = f"meilisearch-macos-{platform_str}"
+        elif system == "linux":
+            if machine in ["x86_64", "amd64"]:
+                platform_str = "amd64"
+            elif machine in ["aarch64", "arm64"]:
+                platform_str = "aarch64"
+            else:
+                raise RuntimeError(f"Unsupported Linux architecture: {machine}")
+            binary_name = f"meilisearch-linux-{platform_str}"
+        else:
+            raise RuntimeError(f"Unsupported platform: {system}")
+
+        # Download the latest release
+        download_url = (
+            f"https://github.com/meilisearch/meilisearch/releases/latest/download/{binary_name}"
```

Comment:
Should we use the `install.meilisearch.com` here and maybe avoid a good amount of code, handling all the potential OS?